idaes/apps/grid_integration/multiperiod/price_taker_model.py

```diff
@@ -0,0 +1,1053 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+    Immediate ToDo list:
+        - Write methods to handle the representative day case
+            - automate cash flow expressions (likely use a 
+              weighted sum for representative days?)
+            - automate startup/shutdown constraints to
+              appropriately address the differences when
+              using representative days
+        - Add documentation to the IDAES readthedocs
+        - Add a warning when costs/revenues are given,
+          but are never used during cashflow construction
+"""
+import pandas as pd
+import numpy as np
+
+have_skl = True
+have_kn = True
+try:
+    from sklearn.cluster import KMeans
+except:
+    have_skl = False
+try:
+    from kneed import KneeLocator
+except:
+    have_kn = False
+
+import matplotlib.pyplot as plt
+
+import os.path
+
+from pyomo.environ import (
+    ConcreteModel,
+    Block,
+    Var,
+    RangeSet,
+    Objective,
+    Constraint,
+    NonNegativeReals,
+    Expression,
+    maximize,
+    Param,
+)
+
+from idaes.apps.grid_integration import MultiPeriodModel
+from idaes.apps.grid_integration.multiperiod.design_and_operation_models import (
+    DesignModelData,
+    OperationModelData,
+)
+
+
+import idaes.logger as idaeslog
+
+_logger = idaeslog.getLogger(__name__)
+
+
+class PriceTakerModel(ConcreteModel):
+    def __init__(self, seed=20, horizon_length=24):
+        super().__init__()
+        self.seed = seed
+        self.horizon_length = horizon_length
+
+    @property
+    def seed(self):
+        return self._seed
+
+    @seed.setter
+    def seed(self, value):
+        if not isinstance(value, int):
```

Comment:
Minor comment, but would it be more efficient to just cast value to an int? I.e.:

```
try:
    self._seed = int(value)
except TypeError as e:
   raise TypeError("My message")
```

Also, I think `TypeError` might be more applicable here.